Clean up refinery and harvester docking sequence logic

penev92 opened the pull request:
This has been bugging me for a long time now, so I finally decided to address it.
@Phrohdoh mentioned a few days ago on IRC that he wants to do some work on refineries, so if this is conflicting with his work we'll just close it for later (if needed), since it's not a feature or a bug fix.
Basically what I did:
- Rename TiberiumRefinery -> TiberiumDawnRefinery (only Tiberian Dawn mod uses it)
- Rename HarvesterDockSequence -> TDHarvesterDockSequence (same reason)
- Rename OreRefinery -> Refinery  (since both RA and Dune 2000 mods uses it)
- Rename RAHarvesterDockSequence -> HarvesterDockSequence  (same reason)
- Fixed a typo in RefineryInfo's DockAngle's description

penev92 committed:
Fix OreRefinery and TiberiumRefinery names

penev92 committed:
Update Refinery docking rules in YAML files

penev92 committed:
Fix harvester dock sequence names

penev92 committed:
Add SpriteHarvesterDockSequence and VoxelHarvesterDockSequence

Pass drag-related info from Refinery to HarvesterDockSequence

penev92 commented:
This became a real mess after all the discussions and change requests. All that time since it's been last touched and the fact that we tried to put things from simple renaming of classes to architectural restructuring and adding new interfaces in one PR didn't help either. So I am taking this back to its original purpose - to fix the bogus class names. No fancy docking logic, no interfaces for it, no rewriting of the Refinery or Harvester classes.
I do intend to still do the docking improvements etc, but in (a) follow-up PR(s). "Small and simple" tends to have a bigger chance of getting merged.

obrakmann commented on OpenRA.Mods.TS/Traits/Buildings/TiberianSunRefinery.cs:
```diff
@@ -11,22 +11,22 @@
 		{
-			return new VoxelHarvesterDockSequence(harv, self);
+			// TODO: Pass DockAngle from YAML
+			return new VoxelHarvesterDockSequence(harv, self, 160, false);
```
Why don't you? You can take the value for the `dragRequired` arg from yaml here, too.

obrakmann commented on OpenRA.Mods.RA/Activities/HarvesterDockSequence.cs:
```diff
@@ -0,0 +1,106 @@
+		static readonly WVec DockOffset = new WVec(-640, 341, 0);
+
+		// TODO: Pass this as RefineryInfo.DragLength in YAML
+		static readonly int DragLength = 12;
```
Here too: why don't you? Doesn't look like such a huge effort to make.

obrakmann commented on OpenRA.Mods.RA/OpenRA.Mods.RA.csproj:
```diff
@@ -80,7 +81,7 @@
     <Compile Include="Activities\LayMines.cs" />
     <Compile Include="Activities\Leap.cs" />
-    <Compile Include="Activities\RAHarvesterDockSequence.cs" />
+    <Compile Include="Activities\HarvesterDockSequence.cs" />
```
Would it be possible to move this to Mods.Common while you're at it?

penev92 replied:
I'll need to look at that before answering. At the time of doing this (the first time) it was impossible. If it requires moving any more classes now I'll leave it for another PR.

penev92 replied:
No, this will stay here for the moment. I've promised to move all the Harvester- and Refinery-related files from Mods.RA to Mods.Common when this is done anyway, so I'll just do it at once.

obrakmann commented:
This will also need an upgrade rule to handle the renames.

Looks ok otherwise, didn't see any issue with the refineries.

penev92 committed:
Changed IAcceptOre to IAcceptResources

penev92 commented:
Updated, rebased and addressed TODOs.
Also removed the `TiberianDawnRefinery` trait, so we now have `Refinery` and `TiberianSunRefinery`, using `SpriteDockSequence` and `VoxelDockSequence` respectively.

penev92 commented:
Updated, rebased and addressed TODOs.
Also removed the `TiberianDawnRefinery` trait, so we now have `Refinery` and `TiberianSunRefinery`, using `SpriteHarvesterDockSequence` and `VoxelHarvesterDockSequence` respectively.
I will look into adding YAML update rules tomorrow.

penev92 committed:
Add YAML upgrade rule

Mailaender commented on OpenRA.Mods.TS/Activities/VoxelHarvesterDockSequence.cs:
```diff
@@ -8,70 +8,34 @@
-	}
-}
+    public class VoxelHarvesterDockSequence : HarvesterDockSequence
+    {
```
Please use tabs instead of spaces.

Mailaender commented:
:+1: No regressions in all 4 mods.

obrakmann commented:
:+1:

obrakmann commented:
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog/_compare/bf2b91ccb81e24b2ffe1c73f896c9aaff6a5cf00%5E...bf2b91ccb81e24b2ffe1c73f896c9aaff6a5cf00)
